fuel/src/main/kotlin/com/github/kittinunf/fuel/toolbox/HttpClient.kt

```diff
@@ -197,45 +211,16 @@ class HttpClient(
         )
     }
 
-    private fun dataStream(request: Request, connection: HttpURLConnection): InputStream? {
-        return try {
-            try {
-                val inputStream = hook.interpretResponseStream(request, connection.inputStream)
-                BufferedInputStream(inputStream)
-            } catch (_: IOException) {
-                // The InputStream SHOULD be closed, but just in case the backing implementation is faulty, this ensures
-                // the InputStream ís actually always closed.
-                try { connection.inputStream?.close() } catch (_: IOException) {}
-
-                connection.errorStream?.let {
-                    BufferedInputStream(hook.interpretResponseStream(request, it))
-                }
-            } finally {
-                // We want the stream to live. Closing the stream is handled by Deserialize
-            }
-        } catch (exception: IOException) {
-            // The ErrorStream SHOULD be closed, but just in case the backing implementation is faulty, this ensures the
-            // ErrorStream ís actually always closed.
-            try { connection.errorStream?.close() } catch (_: IOException) {}
-
-            hook.httpExchangeFailed(request, exception)
-
-            ByteArrayInputStream(exception.message?.toByteArray() ?: ByteArray(0))
-        } finally {
-            // We want the stream to live. Closing the stream is handled by Deserialize
-        }
+    private fun dataStream(request: Request, connection: HttpURLConnection): InputStream? = try {
+        hook.interpretResponseStream(request, connection.inputStream)?.buffered()
+    } catch (_: IOException) {
+        hook.interpretResponseStream(request, connection.errorStream)?.buffered()
```

Comment:
This can throw an IO Exception (see the old line 219), or was that incorrect?

Reply:
if inputStream throws, it will hit errorStream. Also, errorStream doesn't throw.

Reply:
Ok 👍